src/conditional-compilation.md

```diff
@@ -169,11 +169,13 @@ on the endianness of the target's CPU.
 
 ### `target_pointer_width`
 
-Key-value option set once with the target's pointer width in bits. For example,
-for targets with 32-bit pointers, this is set to `"32"`. Likewise, it is set
-to `"64"` for targets with 64-bit pointers.
+Key-value option set once with the target's pointer width in bits.
 
-<!-- Are there targets that have a different bit number? -->
+Example values:
+
+* `"16"`
+* `"32"`
+* `"64"`
```

Comment:
Would this allow targets where pointers are really 24-bits, such as the 65816?
Technically speaking, I forcibly extend pointers to 32-bit, but they only have 24 significant bits.

Reply:
There are two separate questions to answer here, I think.

First, whether pointer size here refers to the "significant bits" or the actual size of a pointer value. I think it refers to the latter, so even if you only have 24 significant bits, if you extend pointers to 32-bit then the pointer size is 32-bit.

But if you *don't* extend pointers, then that raises an interesting question. I don't see a fundamental reason why (for instance) we couldn't support systems with 80-bit pointers and 80-bit `usize`, or why it would *have* to be rounded up to 128-bit, as long as LLVM (or whatever backend we used for the platform) supported those 80-bit values.

If we add mentions of such cases, though, then I think we'd need to change the clarifying note I suggested to say that "Many pieces of Rust code may assume that pointers, `usize`, and `isize` are either 32-bit or 64-bit". Because a system with 40-bit or 48-bit or 56-bit pointers would likely run into many of the same difficulties and limited library support as a system with 16-bit or 24-bit or 80-bit or 128-bit pointes.

Reply:
I extend pointer size to a multiple of the alignment, 4-bytes. However I treat the 4th byte as padding, rather than treat it as part of the value.